engine/compiler/compiler.go

```diff
@@ -339,6 +341,13 @@ func (c *Compiler) Compile(ctx context.Context, args Args) *engine.Spec {
 		if c.Placeholder != "" {
 			dst.Placeholder = c.Placeholder
 		}
+
+		hostnames = append(hostnames, src.Name)
```

Comment:
the only thought I have is that step names are freeform and may not be valid hostnames. what about only appending the hostname if it is a valid hostname? we could use https://godoc.org/github.com/asaskevich/govalidator#IsHost